subscriptions-api.md

```diff
@@ -0,0 +1,613 @@
+infrastructure. This document does not formalize the relationship between the
+original event producer and the subscription manager, and how the subscription
+manager obtains the events it distributes to subscribers.
+
+This document also does not formalize the relationship between a specific set of
+events and the subscription manager. An event producer that produces multiple
+different types of events can offer one subscription manager for all events or a
+subscription manager for each type of event, or any other combination. An event
+producer may also offer up the same set of events for subscription at multiple
+concurrent subscription managers. Such subscription offers are published in a
+CloudEvents Discovery service.
+
+As with the core CloudEvents specification, the goal of this specification is to
+reuse mechanisms based on existing standards and conventions where such exist
+and only introduce new mechanisms where required.
+
+Therefore, this specification not only defines a new subscription management API
+for certain use-cases, but also refers to existing mechanisms available in the
+specifications of transport protocols for which CloudEvents bindings exist.
+
+For instance, introducing and mandating a CloudEvents-specific subscription
+mechanism for MQTT would only complicate implementations while not providing any
+obvious advantage. Using any of the referred native subscription mechanisms of
+the respective transport protocols is therefore within the scope and in
+compliance with this specification.
+
+## 2. Notations and Terminology
+
+### Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+### Terminology
+
+This specification uses the following terms:
+
+#### Source
+
+An event source is a logical entity in a system on behalf of which events are
+produced based on occurrences.
+
+#### Producer
+
+The entity producing the event. The event might reflect an occurrence on a
+source elsewhere in the system. The producer is the concrete entity that creates
+and sends the event related to an occurrence.
+
+#### Intermediary
+
+An "intermediary" (also referred to as middleware) receives an event for the
+purpose of forwarding it to the next receiver, which might be another
+intermediary or a consumer. A typical task for an intermediary is to route the
+event to receivers based on the information in the event context.
+
+#### Consumer
+
+An entity receiving and processing events. Consumers may receive events from
+producers directly or via intermediaries. A consumer might listen and wait for
+events to be delivered to it or it might actively solicit them from the producer
+or intermediary.
+
+#### Subscription
+
+A relationship established between a consumer and a producer or an intermediary.
```

Comment:
In the case of an intermediary, does the consumer actually know it's an intermediary? If not, I wonder if the introduction of the indirection might cause some confusion for some people. Would it be safer to just stick to a 2 party system?

Reply:
While the consumer may not know, if it is a producer or intermediary at the other end, it should not assume one or the other case. A consumer assuming to talk directly to the producer, may create less specific subscriptions and be surprised by all the events it receives upon that subscription.